### PR TITLE
#164337121 Define Tags Model

### PR DESCRIPTION
**Pull request by mrkingz**
#### What does this PR do? 
- defines the Tags model
#### Description of Task to be completed?
- defines the Tags models to create the Tags table and the associated ArticleTags join table
#### What are the relevant pivotal tracker stories?
- [#164337121](https://www.pivotaltracker.com/n/projects/2240406/stories/164337121)

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -41,7 +41,15 @@ app.all("*", (req, res) => {
     message: "Not Found"
   });
 });
-
+if (!isTest) {
+  (async () => {
+    try {
+      await db.sequelize.sync();
+    } catch (error) {
+      console.log("eerr", error);
```
Comment (benfluleck) [inline server/index.js]: What is eerr?

File: server/models/articles.js [hunk 2]
```diff
@@ -83,6 +83,17 @@ const getArticleModel = (sequelize, DataTypes) => {
...
+
+    Article.belongsToMany(db.Tags, {
+      through: {
+        model: "ArticleTags",
+        unique: false
+      },
+      timestamps: false,
+      foreignKey: "articleId",
+      target: "id",
+      onDelete: "RESTRICT"
```
Comment (benfluleck) [inline server/models/articles.js]: Isn't it restricted by default 🤔

File: server/models/tags.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  Tag.associate = db => {
+    Tag.belongsToMany(db.Articles, {
+      through: {
+        model: "ArticleTags",
+        unique: false
+      },
+      timestamps: false,
+      foreignKey: "tagId",
+      target: "id",
+      onDelete: "RESTRICT"
```
Comment (benfluleck) [inline server/models/tags.js]: Isn't this restricted by default

File: server/tests/models/tags.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import models from "@models";
+import { tagData, articleData, userData, category } from "@fixtures";
+
+chai.use(chaiAsPromise);
+const { assert, expect } = chai;
+const { Tags, Categories, Users, Articles } = models;
+
+beforeEach(async () => {
+  await models.sequelize.sync({ force: true });
+});
```
Comment (benfluleck) [inline server/tests/models/tags.spec.js:12]: Please put a space between this block

File: server/tests/models/tags.spec.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+const tagDependencies = async () => {
+  const tagInstance = await Tags.create(tagData[0]);
+  const tagId = tagInstance.get("id");
+	const userInstance = await Users.create(userData[0]);
```
Comment (benfluleck) [inline server/tests/models/tags.spec.js]: What is going on with the indents here ?

File: server/tests/models/tags.spec.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  return Promise.resolve({
+    tagId,
+    articleId,
+    tagInstance,
+    articleInstance
+  });
+};
+
+describe("Tags", () => {
```
Comment (benfluleck) [inline server/tests/models/tags.spec.js]: You are describing the Tags Model not just Tags

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: server/models/articles.js (hunk 2 above)
Comment (mrkingz) [reply server/models/articles.js]: For n:m association, the default for onDelete is CASCADE

File: server/models/tags.js (hunk 3 above)
Comment (mrkingz) [reply server/models/tags.js]: For n:m association, the default for onDelete is CASCADE

File: server/tests/models/tags.spec.js (hunk 5 above)
Comment (mrkingz) [reply server/tests/models/tags.spec.js]: It is well aligned on my system, I guess Github is the one behaving funny.

**Comment by benfluleck** [comment]
I dunno how authRouter is now authsRouter on develop please facilitate the change is this PR please

**Review by obasajujoshua31 — COMMENTED** [review]
please can you add more test to test the association between article and tags. 
@obasajujoshua31, That has been taking care of already. Adding more test, such as you requested, will be redundant. 
Remember you can't add a new record to the join table Articletags  if the instance of Artciles/Tags does not exist.

**Review by eokwukwe — CHANGES_REQUESTED** [review]
Please edit your commit header
`chore(setup): define tags model`
#### should be 
`chore(tagsModel): define tags model`.

It is important that the `<scope>` in the commit header represent exactly what the task is about. Your task is not a `setup` task. The `database` connection has already been setup and running. Your task is to create a `model` for the tags table in the database.

Please try to review the `wiki` for commit messages.

**Comment by mrkingz** [comment]
> Please edit your commit header
> `chore(setup): define tags model`
> 
> #### should be
> `chore(tagsModel): define tags model`.
> 
> It is important that the `<scope>` in the commit header represent exactly what the task is about. Your task is not a `setup` task. The `database` connection has already been setup and running. Your task is to create a `model` for the tags table in the database.
> 
> Please try to review the `wiki` for commit messages.

**Review by eokwukwe — APPROVED** [review]

**Review by obasajujoshua31 — APPROVED** [review]

**Review by benfluleck — APPROVED** [review]

**Pull request merged**